examples/gen_test_data/gen_test_data/run.py

```diff
@@ -0,0 +1,265 @@
+            key: {"connection": value["connection"].format(connection_name=connection_name)}
+            for key, value in CONNECTIONS_TEMPLATE.items()
+        },
+        column_mapping={TEXT_CHUNK: "${data.text_chunk}"},
+        debug=True,
+    )
+    logger.info("Batch run is completed.")
+
+    return base_run
+
+
+def get_batch_run_output(pf: PFClient, base_run: Run):
+    logger.info(f"Start to get batch run {base_run.name} details.")
+    details = pf.get_details(base_run, all_results=True)
+    question = details["outputs.question"].tolist()
+    suggested_answer = details["outputs.suggested_answer"].tolist()
+    debug_info = details["outputs.debug_info"].tolist()
+    return [
+        {"question": q, "suggested_answer": g, "debug_info": d}
+        for q, g, d in zip(question, suggested_answer, debug_info)
+    ]
+
+
+def get_ml_client(subscription_id: str, resource_group: str, workspace_name: str):
+    credential = DefaultAzureCredential(exclude_shared_token_cache_credential=True)
+    return MLClient(
+        credential=credential,
+        subscription_id=subscription_id,
+        resource_group_name=resource_group,
+        workspace_name=workspace_name,
+    )
+
+
+@dsl.pipeline(
+    non_pipeline_inputs=[
+        "flow_yml_path",
+        "should_skip_doc_split",
+        "instance_count",
+        "mini_batch_size",
+        "max_concurrency_per_instance",
+    ]
+)
+def gen_test_data_pipeline(
+    data_input: Input,
+    flow_yml_path: str,
+    connection_name: str,
+    should_skip_doc_split: bool,
+    chunk_size=1024,
+    instance_count=1,
+    mini_batch_size="10kb",
+    max_concurrency_per_instance=2,
+):
+    data = (
+        data_input
+        if should_skip_doc_split
+        else document_split(documents_folder=data_input, chunk_size=chunk_size).outputs.document_node_output
+    )
+    flow_node = load_component(flow_yml_path)(
+        data=data,
+        text_chunk="${data.text_chunk}",
+        connections={
+            key: {"connection": value["connection"].format(connection_name=connection_name)}
+            for key, value in CONNECTIONS_TEMPLATE.items()
+        },
+    )
+
+    flow_node.mini_batch_size = mini_batch_size
+    flow_node.max_concurrency_per_instance = max_concurrency_per_instance
+    flow_node.set_resources(instance_count=instance_count)
+
+    clean_test_data_set(test_data_set_folder=flow_node.outputs.flow_outputs)
+
+
+def run_local(
+    documents_folder,
+    document_chunk_size,
+    document_nodes_file,
+    flow_folder,
+    flow_batch_run_size,
+    connection_name,
+    output_folder,
+    should_skip_split,
+):
+    text_chunks_path = document_nodes_file
+    inner_folder = os.path.join(output_folder, datetime.now().strftime("%b-%d-%Y-%H-%M-%S"))
+    if not os.path.isdir(inner_folder):
+        os.makedirs(inner_folder)
+
+    if not should_skip_split:
+        text_chunks_path = split_document(document_chunk_size, documents_folder, inner_folder)
+
```

Comment:
if should_skip_split:
log: Received document node input file <doc-node-path>, skip step 1 'split documents to doc nodes'.